### ISSUE TITLE
Consider database field to track ODK Central XForm name

### ISSUE THREAD

**Issue by spwoodcock** [hotosm/fmtm#1319]
### EDIT: read comments further down, the idea has changed 

**Describe the bug**

- Various functions require the XForm name in ODK Central to get data.
- After extensive refactoring we finally settled on: `{project_name_prefix}_task_{task_id}`.
- However some old projects may have XForm names in different formats, so data cannot be retrieved from the ODK Central API.

**To Reproduce**

Submissions page of existing prod project, prior to refactor:
https://fmtm.hotosm.org/project-submissions/MTAwMTAw?tab=infographics

**Expected behavior**

- We can access the data for old and new projects, with the XForm name tracked in the database.
- We should probably remove the hardcoded string substituted value for XForm name throughout the code.
- Add a field to the database `public.tasks.xform_name`.
- By default this will be `{project_name_prefix}_task_{task_id}`. We can insert this at the end of new project creation.
- For older projects we can update this to whatever the XForm name is, giving us much more flexibility.
- Then all code should reference: `{tasks.xform_name}` when calling ODK Central.

**Comment by spwoodcock** [comment]
### UPDATE

After discussion we realised it may be possible to duplicate the same xform id if multiple instances of FMTM uses the same ODK Central instance.

- The solution should be to use a UUID for the xform so it's always unique.
- We should update the XForm `name` field to equal `{project_name_prefix}_task_{task_id}`, so that uses in ODK Collect will know what project and task they are mapping.

**Comment by spwoodcock** [comment]
- After implementing entities we now only require one form per project, so this is no longer as much as an issue.
- The xform id should be tracked **per project**.
- As for the concern of duplicated xform ids, first we need to check if different projects can have the same xform id. If xform id only has to be unique **per project**, then this is fine. If it's unique per **odk instance**, then we need to use uuid or similar.

**Comment by Sujanadh** [comment]
so we won't need to create a separate xform name per task?

**Comment by Sujanadh** [comment]
> * After implementing entities we now only require one form per project, so this is no longer as much as an issue.
> * The xform id should be tracked **per project**.
> * As for the concern of duplicated xform ids, first we need to check if different projects can have the same xform id. If xform id only has to be unique **per odk project**, then this is fine. If it's unique per **odk instance**, then we need to use uuid or similar.

i created project with same name twice in odk central. Xform id(task id choices) remained same from the previous project but no addition of task ids from new project.

**Comment by spwoodcock** [comment]
Sorry, can you clarify this?

You managed to create two projects, both with the same xFormId reference?

**Comment by Sujanadh** [comment]
I created a project with certain name and deleted it from database locally because fmtm checks for the project name whether it already exists or not in the database. And i again created the project with same name and checked the central, since the project with same already exists in the central which should have appended the new tasks also but i when checked the form it had only previous project's task ids. I used same entity form but if i use different form with same name it might also not get updated in the central(haven't tested this though).

**Comment by Sujanadh** [comment]
I created a project with certain name and deleted it from database locally because fmtm checks for the project name whether it already exists or not in the database. And I again created the project with the same name and checked the central, since the project with the same name already exists in the central, which should have appended the new tasks also, but when I checked the form, it had only the previous project's task ids. I used the same entity form, but if I use a different form with the same name, it might also not get updated in the central (haven't tested this though).
